### PR TITLE
Pull #675: Fix releasenote-builder: should have execution mode to fail if release version (by pom) is not matching to issues labels

### PR DESCRIPTION
**Pull request by stoyanK7**
Resolves bugs in  #670. Was not sure how to name this commit, so let me know what to rename it to.
Here is the output:
``` 
$ export READ_ONLY_TOKEN=xxx && export DRONE_PULL_REQUEST="master" && ./.ci/releasenotes-gen.sh
```
```
...
HEAD is now at 721550316 Issue #12319: validate PR number when commit is for PR
From https://github.com/checkstyle/checkstyle
 * branch                master     -> FETCH_HEAD
Already up to date.
LATEST_RELEASE_TAG=checkstyle-10.3.4
CS_RELEASE_VERSION=10.3.5
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.

[WARN] Issue #12243 "Pitest/Checker: Drop New/Unnecessary Displays And Switch to Git Diff" is not closed! Please review issue https://github.com/checkstyle/checkstyle/issues/12243
[WARN] Issue #11214 "Specify violation messages in input files." is not closed! Please review issue https://github.com/checkstyle/checkstyle/issues/11214

Validation of release number failed. Release number is a patch(10.3.5), but release notes contain 'new' or 'breaking compatability' labels. Please correct release number by running https://github.com/checkstyle/checkstyle/actions/workflows/bump-version-and-update-milestone.yml

Generation ends with 1 errors.

```

### CODE REVIEW

**Comment by stoyanK7** [comment]
@romani

**Comment by romani** [comment]
Name commit "Pull #675: ......" As description of this pull contains definition bif problem or steps to reproduce.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: releasenotes-builder/src/main/java/com/github/checkstyle/MainProcess.java [hunk 1]
```diff
@@ -95,7 +91,7 @@ public static List<String> runPostGenerationAndPublication(
      * @param cliOptions command line options.
      * @return list of notes validation errors.
      */
-    private static List<String> validateNotes(Multimap<String, ReleaseNotesMessage> releaseNotes,
+    public static Optional<String> validateNotes(Multimap<String, ReleaseNotesMessage> releaseNotes,
```
[the remark this answers is not in the document]
Comment (stoyanK7) [reply releasenotes-builder/src/main/java/com/github/checkstyle/MainProcess.java:94]: We can, but method `addErrors()` has to be implemented for `Result` class. If you are okay with that.

File: releasenotes-builder/src/main/java/com/github/checkstyle/Main.java [hunk 2]
```diff
@@ -113,6 +114,11 @@ private static Result runGithubNotesBuilder(CliOptions cliOptions)
 
         final Result result = NotesBuilder.buildResult(localRepoPath, authToken, remoteRepoPath,
                 startRef, endRef);
+        if (cliOptions.isValidateVersion()) {
```
[the remark this answers is not in the document]
Comment (stoyanK7) [reply releasenotes-builder/src/main/java/com/github/checkstyle/Main.java]: Done.

**Review by romani — APPROVED** [review]
Ok to merge

**Comment by romani** [comment]
Merged without TC , too long queue

**Pull request merged**